cmd/sonobuoy/app/operations/gen.go

```diff
@@ -45,10 +44,10 @@ type templateValues struct {
 }
 
 // GenerateManifest fills in a template with a Sonobuoy config
-func GenerateManifest(cfg GenConfig) ([]byte, error) {
-	mode := cfg.ModeName.Get()
+func (g *GenConfig) GenerateManifest() ([]byte, error) {
```

Comment:
I don't know that I like this. Having a `Config` object have methods to execute feels weird. What does this gain us over the old style?

Reply:
I was planning to expand the number of operations GenConfig struct, such as LoadDefaults.  I could make it more procedural, but every operation would be passing in a GenConfig. 

I was also thinking about the consumer of the library and general UX.  When they include a library, Object[.]F(n) will be auto-discovered vs hunting a function.